runtime/src/test/java/org/evosuite/runtime/util/JavaExecCmdUtilUnixTest.java

```diff
@@ -41,6 +38,7 @@ public void unixNeverNull() {
     assertNotNull(JavaExecCmdUtil.getJavaBinExecutablePath(false));
   }
 
+  @Ignore
```

Comment:
Why?

Reply:
I thought this tests are faulty, because they fail if JAVA_HOME is not set.

Maybe we should test if JAVA_HOME is set first?

For Example:
assertNotNull("JAVA_HOME environment variable is not set.",JAVA_HOME_SYSTEM);

Reply:
Shouldn't this be handled by the @Before initTestEnvironment which sets JAVA_HOME?

Reply:
line 18: 
private static final String JAVA_HOME_SYSTEM = System.getenv("JAVA_HOME");

This should be executed before initTestEnvironment().
Consequently JAVA_HOME_SYSTEM is null.

Leading to assertions in:
50:    assertFalse(StringUtils.isEmpty(JAVA_HOME_SYSTEM));
and
84-85: assertThat(JavaExecCmdUtil.getJavaBinExecutablePath(),
                  IsEqual.equalTo(JAVA_HOME_SYSTEM + SEPARATOR + "bin" + SEPARATOR + "java"));

Reply:
Indeed the JAVA_HOME environment variable may not be set and then this fails.

How about changing it to:

private static final String JAVA_HOME_SYSTEM = System.getProperty("java.home");

...or we could remove the assertions on JAVA_HOME_SYSTEM.

Reply:
Removing the assertion will only fix the issue for this test case.

But  JAVA_HOME_SYSTEM is also part of the only assertion of the unixNewBehavior() test. So removing this assertion, will lead to a test without assertions.

Reply:
So what about System.getProperty, which would make it work even if JAVA_HOME is not set?

(I am not sure I understand this test...)

Reply:
I am not sure I understand the test or the class under test as well.

As far as i can say:
The method tested uses the JAVA_HOME environment variable if exists.
Otherwise it uses the current jvm path. (I dont know why)

So i would suggest to use JAVA_HOME if existing, otherwise java.home:

private static final String JAVA_HOME_SYSTEM = System.getenv("JAVA_HOME") != null ? System.getenv("JAVA_HOME") : System.getProperty("java.home");

Reply:
But if you set the JAVA_HOME environment variable, then the java.home property is initialised with that. So why not simply use java.home directly?

Reply:
No,

System.getenv("JAVA_HOME")

always returns the value of the environment variable, while

System.getProperty("java.home")

returns the path to the JRE running the application.

See:
https://stackoverflow.com/questions/45441516/difference-between-java-home-and-java-home

Reply:
Fine, then let's go with your suggestion